server/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -642,4 +680,14 @@ public final void close() throws IOException {
     protected final boolean isClosed() {
         return closed.get();
     }
+
+    @Override
+    public long getLastModifiedTime() throws IOException {
+        try {
+            return lastModifiedTime.getOrRefresh().lastModifiedTime;
+        } catch (UncheckedIOException e) {
+            // wrapped in the cache and unwrap here
+            throw e.getCause();
```

Comment:
Rather than using a SingleObjectCache, for which you basically do not use the time refresh functionality, I wonder if simply extending the LastModifiedTime record with a similar function .getOrRefresh() would be more straightforward and would also enable throwing the involved IOException directly rather than doing this juggling with the UncheckedIOException.

org.elasticsearch.index.translog.TranslogReader#getLastModifiedTime also has such a simple caching (rather than using the SingleObjectCache).

Reply:
Got it, fixed